lib/SILOptimizer/Utils/Generics.cpp

```diff
@@ -387,16 +389,23 @@ bool ReabstractionInfo::prepareAndCheck(ApplySite Apply, SILFunction *Callee,
   if (CalleeGenericSig)
     InterfaceSubs = CalleeGenericSig->getSubstitutionMap(ParamSubs);
 
+  using namespace OptRemark;
   // We do not support partial specialization.
   if (!EnablePartialSpecialization && InterfaceSubs.hasArchetypes()) {
-    DEBUG(llvm::dbgs() << "    Partial specialization is not supported.\n");
+    REMARK_OR_DEBUG(ORE, [&]() {
+      return RemarkMissed("NoPartialSpecialization", *Apply.getInstruction())
+             << IndentDebug(4) << "Partial specialization is not supported";
```

Comment:
I don't think it makes sense to give this remark.
Partial specialization is currently not done at all (except for the library).

Partial specialization means that the caller function is already generic and not specialized.
I'm not sure if a user would even assume that partial specialization could be done in  such a case.

Reply:
Thanks!  Removed these and added a new missed remark for the case when specialization fails because we don't see the definition of the generic.

Looks good now?